Integrated Gameplay Systems (Prototype)/Assets/Scripts/Locomotion and Survival/WaterMovement.cs

```diff
@@ -6,36 +6,94 @@ public class WaterMovement : State, ILocomotion
         gameManager = _gameManager;
+
+        command1 = new MoveCommand(player, Vector3.forward);
+        command2 = new MoveCommand(player, Vector3.left);
+        command3 = new MoveCommand(player, Vector3.right);
+        command4 = new MoveCommand(player, Vector3.back);
+    }
+
+    public override void FixedUpdate()
+    {
+        elapsed += Time.deltaTime;
+        if (elapsed >= 10f)
+        {
+            elapsed = elapsed % 10f;
+            oxygen.SubstractOxygen(5);
+            Debug.Log(oxygen.currentOxygenLevel);
+        }
+
+        DoMove();
+        CheckTag();
     }
 
     public override void EnableState()
     {
-        gameManager.inputManager.RegisterKeyBinding(KeyCode.W, new MoveCommand(player, Vector3.forward));
-        gameManager.inputManager.RegisterKeyBinding(KeyCode.A, new MoveCommand(player, Vector3.left));
-        gameManager.inputManager.RegisterKeyBinding(KeyCode.S, new MoveCommand(player, Vector3.right));
-        gameManager.inputManager.RegisterKeyBinding(KeyCode.D, new MoveCommand(player, Vector3.back));
+        gameManager.inputManager.RegisterKeyBinding(KeyCode.W, command1);
+        gameManager.inputManager.RegisterKeyBinding(KeyCode.A, command2);
+        gameManager.inputManager.RegisterKeyBinding(KeyCode.S, command3);
+        gameManager.inputManager.RegisterKeyBinding(KeyCode.D, command4);
     }
 
     public override void DisableState()
     {
-        gameManager.inputManager.DeregisterKeyBinding(KeyCode.W, new MoveCommand(player, Vector3.forward));
-        gameManager.inputManager.DeregisterKeyBinding(KeyCode.A, new MoveCommand(player, Vector3.left));
-        gameManager.inputManager.DeregisterKeyBinding(KeyCode.S, new MoveCommand(player, Vector3.right));
-        gameManager.inputManager.DeregisterKeyBinding(KeyCode.D, new MoveCommand(player, Vector3.back));
+        gameManager.inputManager.DeregisterKeyBinding(KeyCode.W, command1);
+        gameManager.inputManager.DeregisterKeyBinding(KeyCode.A, command2);
+        gameManager.inputManager.DeregisterKeyBinding(KeyCode.S, command3);
+        gameManager.inputManager.DeregisterKeyBinding(KeyCode.D, command4);
     }
 
     public void DoMove()
     {
-        Transform camTransform = Camera.main.transform;
-        Vector3 forwardMovement = camTransform.forward * Input.GetAxis("Vertical");
-        Vector3 horizontalMovement = camTransform.right * Input.GetAxis("Horizontal");
-        Vector3 movement = Vector3.ClampMagnitude(forwardMovement + horizontalMovement, 1);
-        player.playerGameObject.transform.Translate(movement * speed * Time.deltaTime, Space.World);
+        Vector3 movement = currentDirection.normalized;
+        player.playerGameObject.transform.Translate(movement * speed * Time.deltaTime);
+        currentDirection = Vector3.zero;
+
+        currentDirection = Vector3.zero;
+    }
+
+    public void AddDirection(Vector3 _direction)
+    {
+        currentDirection += _direction;
+    }
+
+    void CheckTag()
```

Comment:
Wordt deze nog gebruikt?
Volgens mij momenteel niet maar zo ja dan zou het volgens mij nog niet kloppen. Als je niet meer in water bent zou je namelijk moeten switchen naar groundmovement. Hier check je nu of je in water bent en zo ja maak je een new WaterMovement aan, maar je zit al in de watermovement class.

Reply:
Oh dit gaat btw over de gehele Checktag functie van WaterMovement, github laat hier de verkeerde snippet zien